### DISCUSSION TITLE
v4.0.1-rc.1

### DISCUSSION THREAD

**Discussion by jdevcs** [ethereum/web3.js#6030] [Announcements]
#### web3

-   No need for polyfilling nodejs `net` and `fs` modules (#5978)
-   Removed IPC provider dependency, IPC path is no longer viable provider. If you wanna use IPC, please install `web3-providers-ipc` and instantiate provider yourself (#5978)

#### web3-core

-   If a transaction object with a `data` property is passed to `txInputOptionsFormatter`, it will now be replaced with `input` (#5915)
-   The types `TransactionTypeParser` and `TransactionBuilder` are now utilizing the type `Transaction` for the transaction object. (#5993)
-   No need for polyfilling nodejs `net` and `fs` modules (#5978)
-   Removed IPC provider dependency, IPC path is no longer viable provider. If you wanna use IPC, please install `web3-providers-ipc` and instantiate provider yourself (#5978)

#### web3-errors

-   `gasLimit` is no longer accepted as a parameter for `MissingGasError` and `TransactionGasMismatchError, and is also no longer included in error message (#5915)

#### web3-eth

-   `signTransaction` will now return `gas` instead of `gasLimit` for returned transaction object regardless of what property name the provider uses (#5915)
-   `formatTransaction` will now replace `data` transaction property with `input` (#5915)
-   `isTransactionCall` will now check if `value.input` `isHexStrict` if provided (#5915)
-   The functions `defaultTransactionBuilder` and `transactionBuilder` are now utilizing the type `Transaction` for the transaction object. (#5993)

#### web3-eth-accounts

-   Moved @ethereumjs/tx, @ethereumjs/common code to our source code (#5963)
-   The method `signTransaction` returned by `privateKeyToAccount` is now accepting the type `Transaction` for its argument. (#5993)

#### web3-eth-contract

-   `getSendTxParams` will now return `input` instead of `data` in returned transaction parameters object (#5915)
-   `Contract` constructor will now thrown new `ContractTransactionDataAndInputError` if both `data` and `input` are passed in `ContractInitOptions` for `Contract` constructor (#5915)
-   The types `ContractInitOptions`, `NonPayableCallOptions` and `PayableCallOptions` are moved to `web3-types`. (#5993)

#### web3-types

-   `data` property in `TransactionOutput` was renamed to `input` (#5915)
-   The method `signTransaction` inside `Web3BaseWalletAccount` is now utilizing the type `Transaction` for its argument. (#5993)
-   The types `FMT_NUMBER`, `NumberTypes`, `FMT_BYTES`, `ByteTypes`, `DataFormat`, `DEFAULT_RETURN_FORMAT`, `ETH_DATA_FORMAT` and `FormatType` moved from `web3-utils`. (#5993)
-   The types `ContractInitOptions`, `NonPayableCallOptions` and `PayableCallOptions` are moved from `web3-eth-contract`. (#5993)

#### web3-utils

-   The types `FMT_NUMBER`, `NumberTypes`, `FMT_BYTES`, `ByteTypes`, `DataFormat`, `DEFAULT_RETURN_FORMAT`, `ETH_DATA_FORMAT` and `FormatType` moved to `web3-types`. (#5993)

#### web3

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-core

-   Added hybrid build (ESM and CJS) of library (#5904)
-   Added source files (#5956)

#### web3-errors

-   Added hybrid build (ESM and CJS) of library (#5904)
-   Added source files (#5956)

#### web3-eth

-   Added source files (#5956)

#### web3-eth-abi

-   Added hybrid build (ESM and CJS) of library (#5904)
-   Added source files (#5956)

#### web3-eth-accounts

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-eth-contract

-   `input` is now an acceptable property for `ContractInitOptions` in place of `data` (either can be used, but `input` is used withing the `Contract` class) (#5915)
-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-eth-ens

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-eth-iban

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-eth-personal

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-net

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-providers-http

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-providers-ipc

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-providers-ws

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-rpc-methods

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-types

-   Added hybrid build (ESM and CJS) of library (#5904)
-   Added source files (#5956)

#### web3-utils

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)

#### web3-validator

-   Added source files (#5956)
-   Added hybrid build (ESM and CJS) of library (#5904)
-   Added functions `isHexString`, `isHexPrefixed`, `validateNoLeadingZeroes` (#5963)

#### web3-core

-   `getConfig` method from `Web3Config` class, `config` is now public and accessible using `Web3Config.config` (#5950)

#### web3-eth

-   Removed dependencies @ethereumjs/tx, @ethereumjs/common (#5963)

#### web3-eth-abi

-   Removed `formatDecodedObject` function (#5934)

#### web3-eth-contract

-   `data` was removed as a property of `ContractOptions` type (#5915)

#### web3-utils

-   Removed dependencies @ethereumjs/tx, @ethereumjs/common (#5963)

#### web3-eth-ens

-   Bug fix of `checkNetwork` in ENS (#5988)

<hr /><em>This discussion was created from the release <a href='https://github.com/web3/web3.js/releases/tag/v4.0.1-rc.1'>v4.0.1-rc.1</a>.</em>

**Comment by ASISBusiness** [comment]
> ### Changed
> 
> 
> 
> #### web3
> 
> 
> 
> -   No need for polyfilling nodejs `net` and `fs` modules (#5978)
> 
> -   Removed IPC provider dependency, IPC path is no longer viable provider. If you wanna use IPC, please install `web3-providers-ipc` and instantiate provider yourself (#5978)
> 
> 
> 
> #### web3-core
> 
> 
> 
> -   If a transaction object with a `data` property is passed to `txInputOptionsFormatter`, it will now be replaced with `input` (#5915)
> 
> -   The types `TransactionTypeParser` and `TransactionBuilder` are now utilizing the type `Transaction` for the transaction object. (#5993)
> 
> -   No need for polyfilling nodejs `net` and `fs` modules (#5978)
> 
> -   Removed IPC provider dependency, IPC path is no longer viable provider. If you wanna use IPC, please install `web3-providers-ipc` and instantiate provider yourself (#5978)
> 
> 
> 
> #### web3-errors
> 
> 
> 
> -   `gasLimit` is no longer accepted as a parameter for `MissingGasError` and `TransactionGasMismatchError, and is also no longer included in error message (#5915)
> 
> 
> 
> #### web3-eth
> 
> 
> 
> -   `signTransaction` will now return `gas` instead of `gasLimit` for returned transaction object regardless of what property name the provider uses (#5915)
> 
> -   `formatTransaction` will now replace `data` transaction property with `input` (#5915)
> 
> -   `isTransactionCall` will now check if `value.input` `isHexStrict` if provided (#5915)
> 
> -   The functions `defaultTransactionBuilder` and `transactionBuilder` are now utilizing the type `Transaction` for the transaction object. (#5993)
> 
> 
> 
> #### web3-eth-accounts
> 
> 
> 
> -   Moved @ethereumjs/tx, @ethereumjs/common code to our source code (#5963)
> 
> -   The method `signTransaction` returned by `privateKeyToAccount` is now accepting the type `Transaction` for its argument. (#5993)
> 
> 
> 
> #### web3-eth-contract
> 
> 
> 
> -   `getSendTxParams` will now return `input` instead of `data` in returned transaction parameters object (#5915)
> 
> -   `Contract` constructor will now thrown new `ContractTransactionDataAndInputError` if both `data` and `input` are passed in `ContractInitOptions` for `Contract` constructor (#5915)
> 
> -   The types `ContractInitOptions`, `NonPayableCallOptions` and `PayableCallOptions` are moved to `web3-types`. (#5993)
> 
> 
> 
> #### web3-types
> 
> 
> 
> -   `data` property in `TransactionOutput` was renamed to `input` (#5915)
> 
> -   The method `signTransaction` inside `Web3BaseWalletAccount` is now utilizing the type `Transaction` for its argument. (#5993)
> 
> -   The types `FMT_NUMBER`, `NumberTypes`, `FMT_BYTES`, `ByteTypes`, `DataFormat`, `DEFAULT_RETURN_FORMAT`, `ETH_DATA_FORMAT` and `FormatType` moved from `web3-utils`. (#5993)
> 
> -   The types `ContractInitOptions`, `NonPayableCallOptions` and `PayableCallOptions` are moved from `web3-eth-contract`. (#5993)
> 
> 
> 
> #### web3-utils
> 
> 
> 
> -   The types `FMT_NUMBER`, `NumberTypes`, `FMT_BYTES`, `ByteTypes`, `DataFormat`, `DEFAULT_RETURN_FORMAT`, `ETH_DATA_FORMAT` and `FormatType` moved to `web3-types`. (#5993)
> 
> 
> 
> ### Added
> 
> 
> 
> #### web3
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-core
> 
> 
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> -   Added source files (#5956)
> 
> 
> 
> #### web3-errors
> 
> 
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> -   Added source files (#5956)
> 
> 
> 
> #### web3-eth
> 
> 
> 
> -   Added source files (#5956)
> 
> 
> 
> #### web3-eth-abi
> 
> 
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> -   Added source files (#5956)
> 
> 
> 
> #### web3-eth-accounts
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-eth-contract
> 
> 
> 
> -   `input` is now an acceptable property for `ContractInitOptions` in place of `data` (either can be used, but `input` is used withing the `Contract` class) (#5915)
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-eth-ens
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-eth-iban
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-eth-personal
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-net
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-providers-http
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-providers-ipc
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-providers-ws
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-rpc-methods
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-types
> 
> 
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> -   Added source files (#5956)
> 
> 
> 
> #### web3-utils
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> 
> 
> #### web3-validator
> 
> 
> 
> -   Added source files (#5956)
> 
> -   Added hybrid build (ESM and CJS) of library (#5904)
> 
> -   Added functions `isHexString`, `isHexPrefixed`, `validateNoLeadingZeroes` (#5963)
> 
> 
> 
> ### Removed
> 
> 
> 
> #### web3-core
> 
> 
> 
> -   `getConfig` method from `Web3Config` class, `config` is now public and accessible using `Web3Config.config` (#5950)
> 
> 
> 
> #### web3-eth
> 
> 
> 
> -   Removed dependencies @ethereumjs/tx, @ethereumjs/common (#5963)
> 
> 
> 
> #### web3-eth-abi
> 
> 
> 
> -   Removed `formatDecodedObject` function (#5934)
> 
> 
> 
> #### web3-eth-contract
> 
> 
> 
> -   `data` was removed as a property of `ContractOptions` type (#5915)
> 
> 
> 
> #### web3-utils
> 
> 
> 
> -   Removed dependencies @ethereumjs/tx, @ethereumjs/common (#5963)
> 
> 
> 
> ### Fixed
> 
> 
> 
> #### web3-eth-ens
> 
> 
> 
> -   Bug fix of `checkNetwork` in ENS (#5988)
> 
> 
> 
> 
> 
> <hr /><em>This discussion was created from the release <a href='https://github.com/web3/web3.js/releases/tag/v4.0.1-rc.1'>v4.0.1-rc.1</a>.</em>
